src/main/java/cms/gov/madie/measure/utils/GroupPopulationUtil.java

```diff
@@ -1,227 +1,61 @@
-      match = CollectionUtils.isNotEmpty(matchedGroups);
-    }
-    return match;
-  }
-
-  public static boolean isGroupPopulationsMatching(Group originalGroup, Group newGroup) {
-    if (!StringUtils.equals(originalGroup.getScoring(), newGroup.getScoring())) {
-      return false;
-    }
-
-    if (!StringUtils.equals(originalGroup.getPopulationBasis(), newGroup.getPopulationBasis())) {
-      return false;
-    }
-
-    List<Population> originalValidPops = getValidPopulations(originalGroup);
-    List<Population> newValidPops = getValidPopulations(newGroup);
-    boolean populationsMatch = isPopulationMatch(originalValidPops, newValidPops);
-    if (!populationsMatch) {
-      return false;
-    }
-
-    List<MeasureObservation> originalValidObs = getValidObservations(originalGroup);
-    List<MeasureObservation> newValidObs = getValidObservations(newGroup);
-    boolean observationsMatch = isMeasureObservationMatch(originalValidObs, newValidObs);
-    if (!observationsMatch) {
-      return false;
-    }
-
-    List<Stratification> originalValidStras = getValidStratifications(originalGroup);
-    List<Stratification> newValidStras = getValidStratifications(newGroup);
-    boolean stratificationMatch = isStratificationMatch(originalValidStras, newValidStras);
-
-    return stratificationMatch;
-  }
-
-  static List<Population> getValidPopulations(Group group) {
-    if (group != null && !CollectionUtils.isEmpty(group.getPopulations())) {
-      return group.getPopulations().stream()
-          .filter(population -> !StringUtils.isBlank(population.getDefinition()))
-          .collect(Collectors.toList());
-    }
-    return Collections.emptyList();
-  }
-
-  static boolean isPopulationMatch(
-      List<Population> originalValidPops, List<Population> newValidPops) {
-    if (CollectionUtils.isEmpty(originalValidPops) && CollectionUtils.isEmpty(newValidPops)) {
-      return true;
-    } else {
-      if (originalValidPops.size() != newValidPops.size()) {
-        return false;
-      }
-      List<Population> nonMatch =
-          originalValidPops.stream()
-              .filter(
-                  originalPop ->
-                      newValidPops.stream()
-                          .noneMatch(
-                              newPop ->
-                                  newPop
-                                      .getDefinition()
-                                      .equalsIgnoreCase(originalPop.getDefinition())))
-              .toList();
-      if (!CollectionUtils.isEmpty(nonMatch)) {
-        return false;
+  public static void setGroupAndPopulationsDisplayIds(Measure measure, Group group) {
+    String groupNumber = String.valueOf(getGroupNumber(group, measure.getGroups()));
+
+    // set group display id
+    group.setDisplayId("Group_" + groupNumber);
+
+    // set population display id
+    if (!CollectionUtils.isEmpty(group.getPopulations())) {
+      List<Population> ips =
+          group.getPopulations().stream()
```

Comment:
Instead of collecting a list, I think we could just do a count ?

boolean hasMultipleIps = group.getPopulations().stream()
    .filter(pop -> pop.getName().equals(PopulationType.INITIAL_POPULATION))
    .count() > 1;

Reply:
@RohitKandimalla Thanks for the suggestion! Applied.